### DISCUSSION TITLE
Fail to deploy when updating DERIVATIVE fields

### DISCUSSION THREAD

**Discussion by soshizaki** [AntlerVC/firetable#1016] [Support / Q&A] [answered]
Hi support, I have a question.

I am trying to set a derivative field but fail to deploy after updating derivative fields.
Version of rowy run is updated on my workspace.
Please let me know how to resolve this issue. 

I attach a screenshot of log and a screen recording of deployment when I update derivative field.

- screenshot of log
![rowy_deployerror_2022-12-14](https://user-images.githubusercontent.com/84081717/207472026-b3f3355c-1b94-494f-b5ce-cae438f571c8.png)

- screen recording of deployment
https://drive.google.com/file/d/169LppRhRDVnS7I4vs5tk5lzovqN6p4hx/view?usp=share_link

- my rowy run version
<img width="661" alt="image" src="https://user-images.githubusercontent.com/84081717/207472341-a7acb5fb-8829-4fb5-9e44-cbdad600ecf5.png">

Let me know if I need to disclose more information about my issue.
I appreciate your kind support. 

Thanks. Soshi

**Comment by bofeiw** [comment]
Derivative code looks correct. Function config seems to be generated correctly.

@soshizaki In your GCP console, can you go to Cloud Run and open the logs of `rowy-backend`? Usually the reason of the failure can be found in here. Can you see anything useful?

**Reply by soshizaki** [reply]
@bofeiw Thanks for your quick support.
As you mentioned, I checked the logs of rowy-backend in Cloud Run. The screen capture is below. Can you check it? 

![CloudRun_log_capture_2022-12-14](https://user-images.githubusercontent.com/84081717/207504575-f5ad9d55-edc7-4377-a8f1-fe32a45dc4d8.png)

**Reply by soshizaki** [reply]
@bofeiw 
What do you think? If you would like to take a look and let us know what you find. Thank you for your cooperation. Thanks.

**Comment by junshiron** [comment]
Hi @bofeiw,　thanks for the support.
I am a colleague of @soshizaki.
I want to solve this problem.

When I looked at the log, I found that an error occurred at the following command.

`firebase deploy --interactive --project propofm-rowy --only functions`
`yarn deploy --project propofm-rowy --only functions`


But I can't find any clue to the solution from log.

I also looked at the log of Cloud functions (R-"table name" generated from cloud RUN).
The log has the following error and there is no clue to solve it either.


```
   "metrics": {},
   "error": {
     "buildpackId": "",
     "buildpackVersion": "",
     "errorType": "OK",
     "canonicalCode": "OK",
     "errorId": "",
     "errorMessage": ""
    
     ...
```

Do you have any other information for a solution?

**Comment by soshizaki** [comment]
Hi @harinij , May I ask for your help about this discussion???  We cannot deploy for DERIVATIVE fields...

**Comment by bofeiw** [comment]
I don't see anything wrong in your code or in the logs. `firebase deploy` just failed without further info.
Can you try to create a clean project and only deploy the derivative that failed and see if that works?

Also cc @shamsmosowi

**Reply by soshizaki** [reply]
@bofeiw Thanks for your reply. Actually I have already created a clean project after @junshiron failed to deploy on our initial project. My attachment is from a clean new project...  What should we do next ? Initialize our workspace ? Should we try self-hosting rowy?

**Comment by soshizaki** [comment, marked as answer]
@bofeiw We assume this issue is from selection of region of Firebase or Cloud Function. 

Actually we selected asia-northeast1 region when we fail deployment of derivative fields. 
Finally after we change region to us-central1 to create project and install Rowy Run, we success to deploy for derivative fields.
